websites/G/Google Docs/presence.ts

```diff
@@ -1,98 +1,98 @@
 const presence = new Presence({
-  clientId: "630478614894477337"
-});
+    clientId: "630478614894477337"
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+let title: string;
 
-presence.on("UpdateData", () => {
+presence.on("UpdateData", async () => {
   const presenceData: PresenceData = {
-    largeImageKey: "docslogo"
-  };
+      startTimestamp: browsingStamp
+    },
+    privacy = await presence.getSetting("privacy");
   if (window.location.pathname.toLowerCase().includes("/document")) {
     presenceData.largeImageKey = "docslogo";
+    title = document.title.replace("- Google Docs", "");
     if (window.location.pathname.toLowerCase().includes("/edit")) {
-      presenceData.largeImageKey = "docslogo";
-      presenceData.details = "Editing a document:";
-      presenceData.state = document.title.replace("- Google Docs", "");
+      presenceData.details = privacy
+        ? "Editing a document"
+        : "Editing a document:";
+      presenceData.state = privacy ? null : title;
     } else {
-      presenceData.largeImageKey = "docslogo";
-      if (window.location.pathname.toLowerCase() === "/document/u/0/") {
+      if (window.location.pathname.toLowerCase() === "/document/u/0") {
         presenceData.details = "Browsing documents";
       } else {
-        presenceData.largeImageKey = "docslogo";
-        if (window.location.pathname.toLowerCase() === "/document/u/0") {
-          presenceData.details = "Browsing documents";
-        } else {
-          presenceData.largeImageKey = "docslogo";
-          presenceData.details = "Viewing a document:";
-          presenceData.state = document.title.replace("- Google Docs", "");
-        }
+        presenceData.details = privacy
+          ? "Viewing a document"
+          : "Viewing a document:";
+        presenceData.state = privacy ? null : title;
```

Comment:
if !privacy then do Viewing document: you dont need "a"
and dont set state to null, just dont set it

Reply:
bruh moment - yeah idk what happened there because it wasn't a problem before lmao